airflow/jobs/scheduler_job_runner.py

```diff
@@ -1300,9 +1300,16 @@ def _create_dag_runs_dataset_triggered(
                 )
 
     def _should_update_dag_next_dagruns(
-        self, dag: DAG, dag_model: DagModel, total_active_runs: int | None = None, *, session: Session
+        self, dag: DAG, dag_model: DagModel, last_dag_run: DagRun | None = None,
+        total_active_runs: int | None = None, *, session: Session
     ) -> bool:
-        """Check if the dag's next_dagruns_create_after should be updated."""
+        """Check if the dag's next_dagruns_create_after should be updated.
+        If last_dag_run is defined, the update was triggered by a scheduling decision in this DAG run
+        """
+        # If last_dag_run is defined, schedule next only if last_dag_run is finished and was an automated run
+        if last_dag_run and not (last_dag_run.state in State.finished_dr_states and
+           last_dag_run.run_type in [DagRunType.SCHEDULED, DagRunType.BACKFILL_JOB]):
```

Comment:
this is the actual bugfix: previously we didn't check run type so manual runs were messing with scheduling decisions